ext/ForwardDiffGPUArraysCoreExt.jl

```diff
@@ -0,0 +1,62 @@
+module ForwardDiffGPUArraysCoreExt
+
+using GPUArraysCore: AbstractGPUArray
+using ForwardDiff: ForwardDiff, Dual, Partials, npartials, partials
+
+function ForwardDiff.seed!(duals::AbstractGPUArray{Dual{T,V,N}}, x,
+                           seed::Partials{N,V} = zero(Partials{N,V})) where {T,V,N}
+    idxs = collect(ForwardDiff.structural_eachindex(duals, x))
+    duals[idxs] .= Dual{T,V,N}.(x[idxs], Ref(seed))
```

Comment:
I think we should use `view` here to reduce allocations, as in the old implementation (see https://github.com/JuliaDiff/ForwardDiff.jl/pull/472). And given #634, is there some overhead here from using broadcasting instead of `map!`?
```suggestion
    duals[idxs] .= Dual{T,V,N}.(view(x, idxs), Ref(seed))
```